api/src/evaluation/application/smart-random-simulator/index.js

```diff
@@ -0,0 +1,92 @@
+import Joi from 'joi';
+import { securityPreHandlers } from '../../../shared/application/security-pre-handlers.js';
+import { smartRandomSimulatorController } from './smart-random-simulator-controller.js';
+import { AnswerStatus, KnowledgeElement } from '../../../../lib/domain/models/index.js';
+import { LOCALE } from '../../../shared/domain/constants.js';
+import { identifiersType } from '../../../shared/domain/types/identifiers-type.js';
+
+const register = async function (server) {
+  server.route([
+    {
+      method: 'POST',
+      path: '/api/admin/smart-random-simulator',
+      config: {
+        pre: [
+          {
+            method: (request, h) =>
+              securityPreHandlers.hasAtLeastOneAccessOf([
+                securityPreHandlers.checkAdminMemberHasRoleSuperAdmin,
+                securityPreHandlers.checkAdminMemberHasRoleSupport,
+                securityPreHandlers.checkAdminMemberHasRoleMetier,
+                securityPreHandlers.checkAdminMemberHasRoleCertif,
+              ])(request, h),
+            assign: 'hasAuthorizationToAccessAdminScope',
+          },
+        ],
+        validate: {
+          payload: Joi.object({
+            data: {
+              attributes: {
+                'knowledge-elements': Joi.array()
+                  .items({
+                    source: Joi.string()
+                      .valid(KnowledgeElement.SourceType.DIRECT, KnowledgeElement.SourceType.INFERRED)
+                      .required(),
+                    status: Joi.string()
+                      .valid(
+                        KnowledgeElement.StatusType.VALIDATED,
+                        KnowledgeElement.StatusType.INVALIDATED,
+                        KnowledgeElement.StatusType.RESET,
+                      )
+                      .required(),
+                    'answer-id': identifiersType.answerId,
+                    'skill-id': identifiersType.skillId,
+                  })
+                  .min(1)
+                  .required(),
+                answers: Joi.array().items({
+                  result: Joi.string()
+                    .valid(
+                      AnswerStatus.statuses.OK,
+                      AnswerStatus.statuses.KO,
+                      AnswerStatus.statuses.SKIPPED,
+                      AnswerStatus.statuses.FOCUSEDOUT,
+                      AnswerStatus.statuses.PARTIALLY,
```

Comment:
```suggestion
                      ...AnswerStatus.statuses,
```

On pourrait alléger un peu en utilisant le spread non ?

Reply:
On en a longuement débattu. On a hésité. On a dit non. Pour des questions de lisibilité.